### ISSUE TITLE
number reading

### ISSUE THREAD

**Issue by nvdaTrac** [nvaccess/test_nvdaTracImport#258]
**Reported by jrutkowski on 18 Sep 2008 15:15 UTC**
**Blocked by #4273**
A checkbox in the voice dialog box that toggles whether the currently selected synth speaks full numbers of digits.

**Comment by nvdaTrac** [comment]
**Comment by pvagner on 18 Sep 2008 17:47 UTC**
 I am not sure I can understand correctly. Are you asking for an option which will force NVDA to say "one two three four" instead of one thousand two hundret thirty four?
I think this is very experimental feature because it will even make reading times and dates digit by digit. More over this is already possible with speech dictionaries.
Here is an example entry:
Pattern: "(\d)(\d)"
Replacement: "\1 \2 "
regular expression: yes.
Pattern and replacement should be typed without the quotes. This can be used in the voice or temporary dictionary.

**Comment by nvdaTrac** [comment]
**Modified by jteh on 18 Sep 2008 20:49 UTC**

**Comment by nvdaTrac** [comment]
**Comment by aleksey_s on 16 Feb 2009 21:57 UTC**
developers, what do you thing about this one? I offer closing becouse nvda for now does not preprocess numbers (just leading zeros) and all other stuff can be done through dictionaries.

**Comment by nvdaTrac** [comment]
**Comment by jteh on 16 Feb 2009 22:12 UTC**
As noted above, this can currently be done with speech dictionaries. We won't implement this, at least for now.

**Comment by nvdaTrac** [comment]
**Comment by drein on 3 Aug 2011 08:42 UTC**
I re-open the ticket because this feature is quite popular in Italy.
 This feature is very useful for people that must read phone number at work.
The best will be to have a radio button or a combobox that say:
read number as: single digit, 2, 3, normal.
For example, the number 123456789:
we could decide if read as:
1, 2, 3, 4, 5, 6, 7, 8, 9.
12, 34, 56, 78, 9.
123, 456, 789.
Is it difficult to do?

**Comment by nvdaTrac** [comment]
**Comment by nvdakor on 25 Jul 2014 12:56 UTC**
Hi,
This could be done using advanced facilities in speech dictionaries.

**Comment by nvdaTrac** [comment]
**Comment by jteh on 28 Jul 2014 06:38 UTC**
Closing as a duplicate of #4273 because there is more activity and some code there.
